File: solutions/dart/prime-factors/1/lib/prime_factors.dart
// In reverence to Shree DR.MDD — My Infinite Source of Strength

class PrimeFactors {
  List<int> factors(int n) {
    final result = <int>[];
    var div = 2;

    while (div * div <= n) {
      if (n % div == 0) {
        result.add(div);
        n ~/= div;
      } else {
        div++;
      }
    }

    if (n > 1) result.add(n);
    return result;
  }
}

File: solutions/erlang/prime-factors/1/src/prime_factors.erl
% For my Junko F. Didi and Shree DR.MDD

-module(prime_factors).
-export([factors/1]).

factors(X) when not is_integer(X); X =< 0 ->
    exit(badarg);
factors(1) ->
    [];
factors(X) ->
    factors(X, 2, []).

factors(1, _, Acc) ->
    lists:sort(Acc);
factors(Y, Z, Acc) when Y rem Z =:= 0 ->
    factors(Y div Z, Z, [Z|Acc]);
factors(Y, 2, Acc) ->
    factors(Y, 3, Acc);
factors(Y, Z, Acc) ->
    factors(Y, Z + 2, Acc).
